src/reanimated2/interpolation.js

```diff
@@ -1,30 +1,117 @@
 import { Extrapolate } from '../derived/interpolate';
 
+function getVal(config) {
+  'worklet';
+
+  const { type, coef, val, ll, rr, x } = config;
+
+  switch (type) {
+    case Extrapolate.IDENTITY:
+      return x;
+    case Extrapolate.CLAMP:
+      if (coef * val < coef * ll) {
+        return ll;
+      }
+      return rr;
+    case Extrapolate.EXTEND:
+    default:
+      return val;
+  }
+}
+
+function isExtrapolate(value) {
+  'worklet';
+
+  return (
+    value === Extrapolate.EXTEND ||
+    value === Extrapolate.CLAMP ||
+    value === Extrapolate.IDENTITY
+  );
+}
+
+function validateType(type) {
+  'worklet';
+
+  if (!type && type !== undefined) {
+    throw new Error(
+      `Reanimated: config object is not valid please provide valid config, for example:
+       interpolate(value, [inputRange], [outputRange], {
+        extrapolateLeft: 'clamp',
+        extrapolateRight: 'extend',
+      }) or interpolate(value, [inputRange], [outputRange], 'clamp')`
+    );
+  }
+
+  if (
+    type?.constructor === Object &&
+    ((Object.keys(type).length === 2 &&
+      !('extrapolateLeft' in type && 'extrapolateRight' in type)) ||
+      (Object.keys(type).length === 1 &&
+        !('extrapolateLeft' in type || 'extrapolateRight' in type)) ||
+      Object.keys(type).length > 2)
+  ) {
+    throw new Error(
+      `Reanimated: config object is not valid please provide valid config, for example:
+       interpolate(value, [inputRange], [outputRange], {
+        extrapolateLeft: 'clamp',
+        extrapolateRight: 'extend',
+      })`
+    );
+  }
+
+  if (typeof type === 'object') {
+    if ('extrapolateLeft' in type && !isExtrapolate(type.extrapolateLeft)) {
+      throw new Error(
+        `Reanimated: not supported value for "extrapolateLeft" \nSupported values: ["extend", "clamp", "identity"]\n Valid example:
+         interpolate(value, [inputRange], [outputRange], {
+          extrapolateLeft: 'clamp',
+        })`
+      );
+    }
+
+    if ('extrapolateRight' in type && !isExtrapolate(type.extrapolateRight)) {
+      throw new Error(
+        `Reanimated: not supported value for "extrapolateRight" \nSupported values: ["extend", "clamp", "identity"]\n Valid example:
+         interpolate(value, [inputRange], [outputRange], {
+          extrapolateRight: 'clamp',
```

Comment:
I'd like to have this and other error messages as string constants:
```js
const EXTRAPOLATE_RIGHT_ERROR_MSG = `(...)`;
```

Reply:
Done ✅